protocol/messenger.go

```diff
@@ -6397,62 +6397,71 @@ func (m *Messenger) syncDeleteForMeMessage(ctx context.Context, rawMessageDispat
 	})
 }
 
-func (m *Messenger) syncSocialSettings(ctx context.Context, rawMessageDispatcher RawMessageHandler) error {
+func (m *Messenger) syncSocialLinks(ctx context.Context, rawMessageDispatcher RawMessageHandler) error {
 	if !m.hasPairedDevices() {
 		return nil
 	}
 
-	socialLinks, err := m.settings.GetSocialLinks()
+	dbSocialLinks, err := m.settings.GetSocialLinks()
 	if err != nil {
 		return err
 	}
-	for _, link := range socialLinks {
-		syncMessage := &protobuf.SyncSocialLinkSetting{
-			Text:  link.Text,
-			Url:   link.URL,
-			Clock: link.Clock,
-		}
-		encodedMessage, err2 := proto.Marshal(syncMessage)
-		if err2 != nil {
-			return err
-		}
-		err = m.withChatClock(func(chatID string, clock uint64) error {
-			rawMessage := common.RawMessage{
-				LocalChatID:         chatID,
-				Payload:             encodedMessage,
-				MessageType:         protobuf.ApplicationMetadataMessage_SYNC_SOCIAL_LINK_SETTING,
-				ResendAutomatically: true,
-			}
-			_, err = rawMessageDispatcher(ctx, rawMessage)
-			return err
-		})
-		if err != nil {
-			return err
-		}
+
+	dbClock, err := m.settings.GetSocialLinksClock()
+	if err != nil {
+		return err
 	}
-	return nil
+
+	_, chat := m.getLastClockWithRelatedChat()
```

Comment:
if we only need chat, it would be better to use other method rather than `getLastClockWithRelatedChat`